lib/src/Uncertainty/Distribution/IndependentCopulaFactory.cxx

```diff
@@ -58,14 +58,14 @@ Distribution IndependentCopulaFactory::build() const
   return buildAsIndependentCopula().clone();
 }
 
-IndependentCopula IndependentCopulaFactory::buildAsIndependentCopula(const Sample &) const
+IndependentCopula IndependentCopulaFactory::buildAsIndependentCopula(const Sample & sample) const
 {
-  return buildAsIndependentCopula();
+  return IndependentCopula(sample.getDimension());
 }
 
-IndependentCopula IndependentCopulaFactory::buildAsIndependentCopula(const Point &) const
+IndependentCopula IndependentCopulaFactory::buildAsIndependentCopula(const Point & parameters) const
 {
-  return buildAsIndependentCopula();
+  return IndependentCopula(parameters.getDimension());
```

Comment:
Here you have to check that parameters is of size 0 as there is no parameter for the independent copula. It is thus impossible to specify the dimension using the vector of parameters